doc/source/ansible_zuul.rst

```diff
@@ -0,0 +1,289 @@
+:title: Zuul
+
+Zuul
+####
+
+Zuul is a pipeline-oriented project gating system.  It facilitates
+running tests and automated tasks in response to Code Review events.
+
+At a Glance
+===========
+
+:Hosts:
+  * https://dashboard.zuul.ansible.com/
+  * https://uptime.zuul.ansible.com/
+:Windmill:
+  * https://opendev.org/windmill
+  * https://opendev.org/windmill/windmill-config
+:Configuration:
+  * `zuul/main.yaml`
+  * `zuul.d`
+:Projects:
+  * https://opendev.org/zuul/zuul
+  * https://github.com/ansible/ansible-zuul-jobs
+  * https://github.com/ansible/project-config
+:Bugs:
+  * https://storyboard.openstack.org/#!/project/679
+:Resources:
+  * `Zuul Reference Manual <https://docs.openstack.org/infra/zuul>`_
+:Chat:
+  * #zuul on freenode
+
+Overview
+========
+
+The Ansible project uses a number of pipelines in Zuul:
+
+**check**
+  Newly uploaded patchsets enter this pipeline to receive an initial result on
+  GitHub.
+
+**gate**
+  Changes that have been approved by core reviewers are enqueued in
+  order in this pipeline, and if they pass tests, will be merged.
+
+**post**
+  This pipeline runs jobs that operate after each change is merged.
+
+**pre-release**
+  This pipeline runs jobs on projects in response to pre-release tags.
+
+**release**
+  When a commit is tagged as a release, this pipeline runs jobs that
+  publish archives and documentation.
+
+**silent**
+  This pipeline is used for silently testing new jobs.
+
+**experimental**
+  This pipeline is used for on-demand testing of new jobs.
+
+**periodic**
+  This pipeline has jobs triggered on a timer for e.g. testing for
+  environmental changes daily.
+
+Zuul watches events in GitHub and matches those events to the pipelines above.
+If a match is found, it adds the change to the pipeline and starts running
+related jobs.
+
+The **gate** pipeline uses speculative execution to improve
+throughput.  Changes are tested in parallel under the assumption that
+changes ahead in the queue will merge.  If they do not, Zuul will
+abort and restart tests without the affected changes.  This means that
+many changes may be tested in parallel while continuing to assure that
+each commit is correctly tested.
+
+Zuul's current status may be viewed at
+`<https://dashboard.zuul.ansible.com/>`_.
+
+Zuul's configuration is stored in `zuul/main.yaml`.  Anyone
+may propose a change to the configuration by editing that file and
+submitting the change to Gerrit for review.
+
+For the full syntax of Zuul's configuration file format, see the `Zuul
+reference manual <https://docs.openstack.org/infra/zuul>`_.
+
+
+Ansible Zuul Hosts
+==================
+
+.. table:: **Ansible Zuul Hosts**
+
+  =========================================  ========  =================
+  Host                                       Provider  Role
```

Comment:
It would be great to generate this list from the content of `ansible/hosts.yaml`.